assets/js/navigation.js

```diff
@@ -0,0 +1,10 @@
+var h2 = document.querySelectorAll("h2")
+var arrayLength = h2.length;
+var anchorList = document.getElementById("anchor-list");
+for (var i = 0; i < arrayLength; i++) { 
+	var li = document.createElement("LI");
+	var aTag = document.createElement("A");
+	aTag.innerHTML = '<a href="#' + h2[i].id + '">' + h2[i].textContent + '</a>';
```

Comment:
This is generating an <a> within an <a>.  You might need to change the href attribute on the aTag var, rather than setting the string '<a>' in the aTag.innerHTML.